app/controllers/actions/StrideAction.scala

```diff
@@ -29,20 +30,18 @@ import uk.gov.hmrc.play.http.HeaderCarrierConverter
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class StrideAction @Inject() (
-  val authConnector: AuthConnector,
-  appConfig: AppConfig
-)(implicit val executionContext: ExecutionContext)
-    extends ActionRefiner[Request, StrideRequest]
+class StrideAction @Inject() (val authConnector: AuthConnector, strideEnrolmentsConfig: AuthStrideEnrolmentsConfig, @Named("appName") appName: String)(implicit
+  val executionContext: ExecutionContext
+) extends ActionRefiner[Request, StrideRequest]
     with AuthorisedFunctions
     with Logging {
-  private lazy val strideLoginUrl: String = s"${appConfig.strideLoginBaseUrl}/stride/sign-in"
-  private lazy val strideSuccessUrl: String = appConfig.strideSuccessUrl
+  private lazy val strideLoginUrl: String = s"${strideEnrolmentsConfig.strideLoginBaseUrl}/stride/sign-in"
+  private lazy val strideSuccessUrl: String = strideEnrolmentsConfig.strideSuccessUrl
 
   def refine[A](request: Request[A]): Future[Either[Result, StrideRequest[A]]] = {
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    val hasRequiredRole = (appConfig.strideEnrolments fold EmptyPredicate)(_ or _)
+    val hasRequiredRole = (strideEnrolmentsConfig.strideEnrolments fold EmptyPredicate)(_ or _)
```

Comment:
It has been a while since I set up a Stride action, but I recollect joining the role information with AuthProviders(PrivilegedApplication). Has this changed? For examples see https://github.com/hmrc/stride-demo-frontend.

Reply:
It doesn't look like AuthProviders(PrivilegedApplication) does very much though. It just indicates that the auth provider is a privileged application rather than the government gateway for example.

Reply:
Yes it is worth adding this predicate, the auth provider type PrivilegedApplication for a stride user. The session is created via stride-auth service here...  https://github.com/hmrc/stride-auth/blob/a7b9b1c88b23a5d9d0a5de2e5b7b3b6fbf16df23/app/models/UserSession.scala#L15

Reply:
I don't think this configuration is working. You need something like

AuthProviders(PrivilegedApplication) and strideEnrolmentsConfig.strideEnrolments.head or strideEnrolmentsConfig.strideEnrolments.tail

but generalised for any number of enrolments.

Reply:
fixed this